public/index.js

```diff
@@ -3,63 +3,59 @@ const buttons = document.querySelectorAll("button");
 const darkLightButton = document.querySelector(".dark-light-button");
 const allergies = document.querySelector(".allergies");
 const title = document.querySelector("h1");
+const loadingContainer = document.getElementById("loading-container");
```

Comment:
There are different ways to access elements, and `getElementById()` is one of them. But since we started using `querySelector()` from the beginning, I think we should stick to it for consistency's sake :-). That would also make our codebase look cleaner. `document.querySelector("#loading-container")` should do the trick.